### PR TITLE
Release 1.21.0

### PR DESCRIPTION
**Pull request by khorne3**
## Release Checklist

- [x] All Feature documentation merged
- [x] Changelog completed
- [x] Technical Engineer Review
- [x] Documentation Owner Review
- [x] Release Manager Review
- [ ] Commit cleaned up and co-authored
- [ ] Create branch for prior version (`1.20.0`)

### CODE REVIEW

File: admin/satellites/management.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  devurlsHost: "*.eu-west.example.com"
+  tls:
+    hostSecretName: "coder-satellite-cert"
+    devurlsHostSecretName: "coder-satellite-cert"
+```
+
+Once you've updated your Helm chart, install the satellite:
+
+```console
+helm add coder https://helm.coder.com
```
Comment (Emyrk) [inline admin/satellites/management.md:78]:
```suggestion
helm repo add coder https://helm.coder.com
```

**Review by bpmct — CHANGES_REQUESTED** [review]
I noticed a few things in our current docs that need to be updated for 1.21. Tracking these here and can submit PRs to this branch to resolve if nobody beats me to it :)

- [x] everything that mentions cemanager is now coderd (ex. the reset password command)
- [x] places that mentions helm values should probably be evaluated (e.x SSH access is now enabled/disabled in the workspace providers page, not helm)
- [x] install page says to look for ingress-nginx IP. new installs on 1.21 do not include an ingress

I'm sure there are also more, I did not do a comprehensive search. These are just some things I've observed using the RC for various things

**Comment by khorne3** [comment]
needs https://github.com/cdr/docs/pull/512 and https://github.com/cdr/docs/pull/511

File: changelog/1.21.0.md [hunk 2]
```diff
@@ -5,7 +5,8 @@ description: "Released on 21/07/2021"
 
 ### Breaking changes ❗
 
-There are no breaking changes in 1.21.0.
+- web: The workspace provider ping functionality has been removed from the Coder
+  UI.
```
Comment (deansheather) [inline changelog/1.21.0.md:9]: This is not a breaking change, legacy workspace providers still have the ping indicator. The ping indicator is only missing for net v2 enabled providers.

File: changelog/1.21.0.md [hunk 3]
```diff
@@ -48,6 +49,19 @@ There are no breaking changes in 1.21.0.
   and maximum disk size constraint to 8 TiB.
 - web: Added ability to choose the SSH keygen algorithm Coder uses when
   generating SSH keys.
+- infra: Added connection cache based on open DataChannels.
```
Comment (deansheather) [inline changelog/1.21.0.md:53]:
```suggestion
```

File: changelog/1.21.0.md [hunk 4]
```diff
@@ -48,6 +49,19 @@ There are no breaking changes in 1.21.0.
...
+- infra: Added option to opt-in to OIDC's new refresh feature, allowing
+  Coder-issued API keys to inherit session timing limits and to use refresh
+  tokens to ensure continued access.
+- infra: Improved speed of metrics query to improve metrics page performance.
+- infra: Added `coderd.trustProxyIP` to Helm chart to allow configuration of
+  Real IP.
+- web: Added ability to enable/disable SSH access to the edit workspace
```
Comment (deansheather) [inline changelog/1.21.0.md:60]: This is called "External Connect" and blocks SSH and external tunneling (except for devurls)

File: changelog/1.21.0.md [hunk 5]
```diff
@@ -48,6 +49,19 @@ There are no breaking changes in 1.21.0.
...
+  providers page.
+- infra: Added functionality so Coder agent output is written to logs.
```
Comment (deansheather) [inline changelog/1.21.0.md:62]:
```suggestion
```

File: changelog/1.21.0.md [hunk 6]
```diff
@@ -48,6 +49,19 @@ There are no breaking changes in 1.21.0.
...
+- web: Images tags now update asynchronously.
+- infra: Added WebRTC service (including ICE server routes)
+- web: Removed appProtocol from `coderd`.
```
Comment (deansheather) [inline changelog/1.21.0.md:64]:
```suggestion
```

File: changelog/1.21.0.md [hunk 7]
```diff
@@ -59,6 +73,11 @@ There are no breaking changes in 1.21.0.
   incorrect.
 - web: Fixed issue where user deletion took an extended period of time (users
   will now be deleted faster).
+- web: Fixed issue with envproxy routes to resolve broken PWA bookmarks.
```
Comment (deansheather) [inline changelog/1.21.0.md:76]: This should be a feature (or just removed entirely)
```
- web: Redirect legacy envproxy routes to the new NetworkingV2 routes to keep bookmarks.
```

**Review by deansheather — APPROVED** [review]

**Pull request merged**